pkg/watch/manager.go

```diff
@@ -241,8 +255,10 @@ func (wm *Manager) replaceWatches(r *Registrar) error {
 
 	// Add desired watches. This is idempotent for existing watches.
 	for gvk := range desired {
-		if err := wm.doAddWatch(r, gvk); err != nil {
-			errlist = append(errlist, fmt.Errorf("adding watch for %+v %w", gvk, err))
+		for r := range desired[gvk].registrars {
```

Comment:
Why has this changed to iterate over all registrars when it is called from a specific registrar?

Reply:
`desired` spans all registrars, so this change fixes a bug where if `r1` desires something, and we call Replace for `r2`, `r1`'s desired roster bleeds into `r2`.

If you're asking whether this could changed to:

```
for gvk := range desired {
        if _, ok := desired[gvk].registrars[r]; !ok {
                continue
        }
	if err := wm.doAddWatch(r, gvk); err != nil {
		errlist = append(errlist, fmt.Errorf("adding watch for %+v %w", gvk, err))
	}
}
```

I *think* it could - I'll squint at it a bit more. Although it might be safer to leave as is, where this function will definitely reconcile to the desired state.
Let me know what you think!

Regarding the original bug, I'll add a test case that captures it.

Reply:
@maxsmythe I've reworked this fix as a separate issue: #620.